examples/tutorials/inset.py

```diff
@@ -0,0 +1,116 @@
+"""
+Create an inset figure
+======================
+
+Plotting an inset figure inside another figure can be done with the
+:meth:`pygmt.Figure.inset` method. After a large figure as been created,
+``inset`` is called using a ``with`` statement to add elements to the inset
+figure.
+"""
+
+import pygmt
+
+########################################################################################
+#
+# Prior to creating an inset figure, a larger figure must first be plotted. In the
+# example below, :meth:`pygmt.Figure.coast` is used to create a map of the US state of
+# Massachusetts.
+
+fig = pygmt.Figure()
+fig.coast(
+    # Sets the region plotted in the larger figure
+    region=[-74, -69.5, 41, 43],
+    # Sets the state boundaries to be plotted with thin lines
+    borders="2/thin",
+    # Plots all shorelines with thin lines
+    shorelines="thin",
+    # Sets a 15 centimeter figure showing a Mercator projection
+    projection="M15c",
+    # Sets the color of the land to "lightyellow"
+    land="lightyellow",
+    # Sets the color of the water to "lightblue"
+    water="lightblue",
+    # Sets the frame to use the automatic settings
+    frame="a",
+)
+fig.show()
+
+########################################################################################
+#
+# The :meth:`pygmt.Figure.inset` method use a context manager, and is called using a
+# ``with`` statement. It does not require any arguments, but will not display an inset
+# figure without a ``location`` argument. Both the location and width of the inset must
```

Comment:
> It does not require any arguments, but will not display an inset figure without a ``location`` argument.

It's a little confusing. I thought `location` (**-D**) is required, no?

Reply:
Not sure about the terminology to use. An error  won't be raise if **-D** is left out (based upon my limited testing) but it won't plot.

Reply:
Looks like a bug or a missing feature to me. Reported to upstream at https://github.com/GenericMappingTools/gmt/issues/4713.

Reply:
Okay. I'll leave it as is for now.

Reply:
We should say that `location` is a required argument.

Reply:
Just pushed the change.